includes/class-wc-admin-reports-segmenting.php

```diff
@@ -363,15 +363,25 @@ protected function set_all_segments() {
 			// 1 -- returning customer
 			$segments = array( 0, 1 );
 		} elseif ( 'tax_rate_id' === $this->query_args['segmentby'] ) {
-			// @todo Do we need to include tax rates that existed in the past, but have never been used? I guess there are other, more pressing problems...
-			// Current tax rates UNION previously used tax rates.
-			$tax_rate_ids = $wpdb->get_results(
-				"SELECT tax_rate_id FROM {$wpdb->prefix}woocommerce_tax_rates
-						UNION 
-						SELECT DISTINCT meta_value FROM {$wpdb->prefix}woocommerce_order_itemmeta where meta_key='rate_id'",
+			$taxes = $wpdb->get_results(
```

Comment:
I think we should consider refactoring the two queries still present in `set_all_segments`. This and coupons above use direct queries. We should instead use a data store method to make swapping things out easier. (This can be a separate `@todo`/issue/PR).

(I think even `get_categories` should be updated, since CPT access in core is also behind a data store too). We want to be using filterable functions like `wc_get_products` or data store methods and not directly touching the database or post types.